### PR TITLE
NSPR-1387 : Xrt hbm support

### PR DESCRIPTION
**Pull request by saifuddin-xilinx**
In this check-in we have following fixes:

- Coverity issue

- xclbin group mapping : While constructing "xclbin_data_sections", we were using **sort** function to sort memory Banks based on address. Which was giving inconsistent sorting data. We have used **stable_sort**. Now this is giving corrent sorting data. 

- xocl driver fix : Xocl driver was using MEM_TOPOLOGY  instead of GROUP_TOPOLOGY for calculating number of memory available.

- xbutil : Re Formatted query and top command output based on memory grouping.

### CODE REVIEW

File: src/runtime_src/xocl/xclbin/xclbin.cpp [hunk 1]
```diff
@@ -726,8 +726,8 @@ class xclbin_data_sections
         // first element not part of the sorted (decreasing) range
         auto upper = std::find_if(itr, m_membanks.end(), [addr] (auto& mb) { return mb.base_addr < addr; });
 
-        // find first used memidx if any, default to first memidx in range if unused
-        auto used = std::find_if(itr, upper, [](auto& mb) { return mb.used; });
+        // find last used memidx if any, default to first memidx in range if unused
+	auto used = std::find_end(itr, upper, itr, itr+1, [](auto& mb, auto& mb2) { (void)mb2; return mb.used;});
```
Comment (chvamshi-xilinx) [inline src/runtime_src/xocl/xclbin/xclbin.cpp:730]: Can you simplify this logic by getting "upper". Just get the "upper" value by checking both addr and size

Comment (saifuddin-xilinx) [reply src/runtime_src/xocl/xclbin/xclbin.cpp]: Updated the logic.

File: src/runtime_src/core/pcie/tools/xbutil/xbutil.h [hunk 2]
```diff
@@ -1632,8 +1562,6 @@ class device {
         for(int32_t i = 0; i < map->m_count; i++) {
             if(map->m_mem_data[i].m_type == MEM_STREAMING)
                 continue;
-            if(!strncmp((const char*)map->m_mem_data[i].m_tag, "HOST", 4))
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/pcie/tools/xbutil/xbutil.h:1635]: This looks like  a merging issue.  Please revert this change

Comment (saifuddin-xilinx) [reply src/runtime_src/core/pcie/tools/xbutil/xbutil.h:1635]: Fixed this issue. Thanks.

**Comment by chvamshi-xilinx** [comment]
unrelated hw_emu tests failure

**Comment by chvamshi-xilinx** [comment]
retest this please
